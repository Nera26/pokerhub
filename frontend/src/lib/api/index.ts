import { createApiClient } from './factory';
import type { paths } from '@contracts/api';

export const client = createApiClient();

export type DefaultAvatarResponse =
  paths['/config/default-avatar']['get']['responses']['200']['content']['application/json'];

export async function fetchDefaultAvatar(): Promise<DefaultAvatarResponse> {
  const { data } = await client.GET('/config/default-avatar');
  return data!;
}

export type PendingWithdrawalsResponse =
  paths['/admin/withdrawals']['get']['responses']['200']['content']['application/json'];

export async function fetchPendingWithdrawals(
  opts: { signal?: AbortSignal } = {},
): Promise<PendingWithdrawalsResponse> {
  const { data } = await client.GET('/admin/withdrawals', {
    signal: opts.signal,
  });
  return data!;
}
