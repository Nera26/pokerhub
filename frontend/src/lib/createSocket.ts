import type { Socket } from 'socket.io-client';
import { getSocket as baseGetSocket, disconnectSocket } from '@/app/utils/socket';

export function createNamespaceSocket(ns: string) {
  let socket: Socket | null = null;

  const getSocket = (
    options: Parameters<typeof baseGetSocket>[0] = {},
  ): Socket => {
    socket = baseGetSocket({ namespace: ns, ...options }) as Socket;
    return socket;
  };

  const disconnect = (): void => {
    if (socket) {
      disconnectSocket(ns);
      socket = null;
    }
  };

  return { getSocket, disconnect };
}

