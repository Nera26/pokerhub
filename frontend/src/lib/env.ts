import { z } from 'zod';

export const IS_E2E = process.env.NEXT_PUBLIC_E2E === '1';

const envSchema = z
  .object({
    NODE_ENV: z
      .enum(['development', 'test', 'production'])
      .default('development'),
    NEXT_PUBLIC_SOCKET_URL: z.string().url().optional(),
    NEXT_PUBLIC_BASE_URL: z.string().url().optional(),
    NEXT_PUBLIC_E2E: z.string().optional(),
    VERCEL_URL: z.string().optional(),
    CI: z.string().optional(),
    NEXT_PUBLIC_LOG_LEVEL: z
      .enum(['error', 'warn', 'info', 'debug'])
      .default('info'),
  })
  .transform((env) => ({
    ...env,
    NEXT_PUBLIC_SOCKET_URL:
      env.NEXT_PUBLIC_SOCKET_URL ?? 'http://localhost:4000',
    NEXT_PUBLIC_BASE_URL: env.NEXT_PUBLIC_BASE_URL ?? 'http://localhost:3000',
    IS_E2E: env.NEXT_PUBLIC_E2E === '1',
  }));

const parsed = envSchema.safeParse(process.env);

if (!parsed.success) {
  const errors = parsed.error.errors
    .map((err) => `${err.path.join('.')}: ${err.message}`)
    .join('\n');
  throw new Error(`Invalid environment variables:\n${errors}`);
}

export const env = parsed.data;

export type Env = typeof env;
