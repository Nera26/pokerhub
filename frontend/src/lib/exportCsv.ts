export function exportCsv(
  filename: string,
  header: string[],
  rows: string[][],
): void {
  const csv = [header, ...rows].map((r) => r.join(',')).join('\n');
  const blob = new Blob([csv], { type: 'text/csv;charset=utf-8;' });
  const url = URL.createObjectURL(blob);
  const link = document.createElement('a');
  link.href = url;
  link.download = filename;
  link.click();
  URL.revokeObjectURL(url);
}
