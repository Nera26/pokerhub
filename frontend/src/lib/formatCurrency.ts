const formatterCache = new Map<string, Intl.NumberFormat>();
const FALLBACK_CURRENCY = 'USD';

function createFormatter(currency: string) {
  return new Intl.NumberFormat(undefined, {
    style: 'currency',
    currency,
    minimumFractionDigits: 2,
    maximumFractionDigits: 2,
  });
}

function normalizeCurrency(currency?: string) {
  return (currency ?? FALLBACK_CURRENCY).toUpperCase();
}

export function getCurrencyFormatter(currency?: string): Intl.NumberFormat {
  const normalized = normalizeCurrency(currency);
  const cached = formatterCache.get(normalized);
  if (cached) {
    return cached;
  }

  try {
    const formatter = createFormatter(normalized);
    formatterCache.set(normalized, formatter);
    return formatter;
  } catch {
    if (normalized !== FALLBACK_CURRENCY) {
      return getCurrencyFormatter(FALLBACK_CURRENCY);
    }

    const fallbackCached = formatterCache.get(FALLBACK_CURRENCY);
    if (fallbackCached) {
      return fallbackCached;
    }

    const fallbackFormatter = createFormatter(FALLBACK_CURRENCY);
    formatterCache.set(FALLBACK_CURRENCY, fallbackFormatter);
    return fallbackFormatter;
  }
}

export interface FormatCurrencyOptions {
  signed?: boolean;
}

export function formatCurrency(
  amount: number,
  currency?: string,
  options?: FormatCurrencyOptions,
): string {
  const formatter = getCurrencyFormatter(currency);
  const normalizedAmount = Object.is(amount, -0) ? 0 : amount;
  const formatted = formatter.format(normalizedAmount);

  if (options?.signed && normalizedAmount > 0) {
    return `+${formatted}`;
  }

  return formatted;
}
