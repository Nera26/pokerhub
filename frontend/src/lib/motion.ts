import type { Transition } from 'framer-motion';

export const m: Record<'fast' | 'base' | 'slow', Transition> = {
  fast: { duration: 0.08, ease: 'easeOut' },
  base: { duration: 0.12, ease: 'easeOut' },
  slow: { duration: 0.2, ease: 'easeOut' },
};
