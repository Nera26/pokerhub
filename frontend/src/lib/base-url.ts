import { env } from './env';

export function getBaseUrl() {
  if (env.NEXT_PUBLIC_BASE_URL) {
    return env.NEXT_PUBLIC_BASE_URL;
  }
  if (env.VERCEL_URL) {
    return `https://${env.VERCEL_URL}`;
  }
  return 'http://localhost:3000';
}
