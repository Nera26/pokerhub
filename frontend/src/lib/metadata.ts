import { getBaseUrl } from './base-url';

const DEFAULT_TITLE = 'PokerHub';
const DEFAULT_DESCRIPTION =
  "Live Texas Hold'em, Omaha & Tournaments â€” PokerHub";
const DEFAULT_IMAGE = '/pokerhub-logo.svg';

interface SiteMetadata {
  title: string;
  description: string;
  imagePath: string;
}

let metadataPromise: Promise<SiteMetadata> | null = null;

async function getSiteMetadata(): Promise<SiteMetadata> {
  if (!metadataPromise) {
    metadataPromise = fetch('/api/site-metadata')
      .then(async (res) => {
        if (res.ok) {
          return (await res.json()) as SiteMetadata;
        }
        throw new Error('Failed to fetch');
      })
      .catch(() => ({
        title: DEFAULT_TITLE,
        description: DEFAULT_DESCRIPTION,
        imagePath: DEFAULT_IMAGE,
      }));
  }
  return metadataPromise;
}

interface BuildMetadataOptions {
  title?: string;
  description?: string;
  imagePath?: string;
  path?: string;
}

export async function buildMetadata({
  title,
  description,
  imagePath,
  path = '',
}: BuildMetadataOptions = {}) {
  const baseUrl = getBaseUrl();
  const meta = await getSiteMetadata();
  return {
    title: title ?? meta.title,
    description: description ?? meta.description,
    image: `${baseUrl}${imagePath ?? meta.imagePath}`,
    url: `${baseUrl}${path}`,
  };
}
