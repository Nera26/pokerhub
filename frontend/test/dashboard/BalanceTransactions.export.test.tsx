import { exportCsv } from '@/lib/exportCsv';

jest.mock('@/lib/exportCsv', () => ({ exportCsv: jest.fn() }));

describe('BalanceTransactions CSV export', () => {
  it('generates expected CSV', () => {
    const log = [
      {
        datetime: '2024-01-01',
        action: 'Add Balance',
        amount: 100,
        by: 'admin',
        notes: 'note',
        status: 'Completed',
      },
    ];
    const header = [
      'Date & Time',
      'Action',
      'Amount',
      'Performed By',
      'Notes',
      'Status',
    ];
    const rows = log.map((t) => [
      t.datetime,
      t.action,
      `+${t.amount}`,
      t.by,
      `"${t.notes.replace(/"/g, '""')}"`,
      t.status,
    ]);
    const csv = [header, ...rows].map((r) => r.join(',')).join('\n');
    expect(csv).toMatchSnapshot();
  });
});
