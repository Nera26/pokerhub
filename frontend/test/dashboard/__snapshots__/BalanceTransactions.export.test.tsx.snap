// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BalanceTransactions CSV export generates expected CSV 1`] = `
"Date & Time,Action,Amount,Performed By,Notes,Status
2024-01-01,Add Balance,+100,admin,"note",Completed"
`;
